### PR TITLE
Feature/load script action

### LINKED ISSUE #201: LOAD_SCRIPT action
## Description
Scripts should be able to be loaded using an action (as discussed here https://github.com/Studio-Lovelies/GG-JointJustice-Unity/discussions/170)

The action should unload the current narrative script and its associated assets, and load the new narrative script.

...

## Acceptance Criteria

This story will be finished when scripts can be loaded with an action

### PR DESCRIPTION
**Pull request by PhoebeMitchell**
## Summary

Adds a LOAD_SCRIPT action which changes the active script played by the NarrativeScriptPlayerComponent, and loads and unloads new BGScenes and old BGScenes respectively.

This change means we don't need to set sequences of narrative scripts in the unity editor.

## Testing instructions

Run the associated tests. Run "LoadScript - Test Scene". Try loading some scripts with the LOAD_SCRIPT action.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource: 
  
closes #201

### CODE REVIEW

File: unity-ggjj/Assets/Scripts/NarrativeScript/NarrativeScript.cs [hunk 1]
```diff
@@ -36,6 +36,10 @@ public NarrativeScript(TextAsset script, IActionDecoder actionDecoder = null)
     /// <param name="actionDecoder">An optional action decoder, used for testing</param>
     public void Initialize(IActionDecoder actionDecoder = null)
     {
+        if (Script == null)
+        {
+            throw new NullReferenceException("Could not initialize narrative script. Script field is null.");
+        }
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Scripts/NarrativeScript/NarrativeScript.cs:42]: You know I like my early outs! Good catch! 👍

**Review by ViMaSter — APPROVED** [review]
A new feature and more coverage. Nothing else to add here, looks good to me! 👍

File: unity-ggjj/Assets/Scripts/NarrativeScript/NarrativeScriptPlaylist/NarrativeScriptPlaylist.cs [hunk 2]
```diff
@@ -1,47 +1,46 @@
...
     [field: Tooltip("A narrative script to be played on game over")]
     [field: SerializeField] public NarrativeScript GameOverScript { get; private set; }
-    
-    private int _narrativeScriptIndex = -1;
 
     /// <summary>
     /// Call the initialise method on all narrative scripts in this playlist
     /// </summary>
-    public void InitializeNarrativeScripts()
+    public void InitializeNarrativeScripts(BGSceneList bgSceneList)
```
Comment (notexplosive) [inline unity-ggjj/Assets/Scripts/NarrativeScript/NarrativeScriptPlaylist/NarrativeScriptPlaylist.cs:24]: channeling Amy when I say this: ✨missing summary for parameter

**Review by notexplosive — APPROVED** [review]
teeny lil nit

**Pull request merged**